src/System/Console/GetOpt/Generics/Internal.hs

```diff
@@ -24,8 +24,13 @@ mapFieldInfo f info = case info of
       cons@Infix{} -> cons
       cons@Constructor{} -> cons
 
-slugify :: String -> String
-slugify [] = []
-slugify (x : xs)
-  | isUpper x = '-' : toLower x : slugify xs
-  | otherwise = x : slugify xs
+normalizeFieldName :: String -> String
+normalizeFieldName =
+    slugify . filter (\ c -> (isAscii c && isAlpha c) || (c == '-'))
```

Comment:
I thought there would be no valid haskell field names without ascii letters in them, but `_` is a valid name. I guess in this case an error should be thrown.